addons/binding/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/discovery/DenonMarantzDiscoveryParticipant.java

```diff
@@ -0,0 +1,163 @@
+import org.openhab.binding.denonmarantz.DenonMarantzBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Jan-Willem Veldhuis
+ *
+ */
+public class DenonMarantzDiscoveryParticipant implements MDNSDiscoveryParticipant {
+
+    private Logger logger = LoggerFactory.getLogger(DenonMarantzDiscoveryParticipant.class);
+
+    // Service type for 'Airplay enabled' receivers
+    private static final String RAOP_SERVICE_TYPE = "_raop._tcp.local.";
+
+    /**
+     * Match the serial number, vendor and model of the discovered AVR.
+     * Input is like "0006781D58B1@Marantz SR5008._raop._tcp.local."
+     * A Denon AVR serial (MAC address) starts with 0005CD
+     * A Marantz AVR serial (MAC address) starts with 000678
+     */
+    private static final Pattern DENON_MARANTZ_PATTERN = Pattern
+            .compile("^((?:0005CD|000678)[A-Z0-9]+)@(.+)\\._raop\\._tcp\\.local\\.$");
+
+    /**
+     * Denon AVRs have a MAC address / serial number starting with 0005CD
+     */
+    private static final String DENON_MAC_PREFIX = "0005CD";
+
+    /**
+     * Marantz AVRs have a MAC address / serial number starting with 000678
+     */
+    private static final String MARANTZ_MAC_PREFIX = "000678";
+
+    private String vendor;
+
+    private String model;
+
+    private String serial;
+
+    private String friendlyName;
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(THING_TYPE_AVR);
+    }
+
+    @Override
+    public String getServiceType() {
+        return RAOP_SERVICE_TYPE;
+    }
+
+    @Override
+    public DiscoveryResult createResult(ServiceInfo serviceInfo) {
+        logger.debug("AVR found: {}", serviceInfo.getQualifiedName());
+        ThingUID uid = getThingUID(serviceInfo);
+        if (uid != null) {
+            Map<String, Object> properties = new HashMap<>(2);
+
+            if (serviceInfo.getHostAddresses().length == 0) {
+                logger.debug("Could not determine IP address for the Denon/Marantz AVR");
+                return null;
+            }
+            String host = serviceInfo.getHostAddresses()[0];
+
+            logger.debug("IP Address: {}", host);
+
+            // try a HTTP request to autoconfigure the HTTP / Telnet parameter
+            HttpClient httpClient = new HttpClient();
+            ContentResponse response;
+            boolean telnetEnable = true;
+            try {
+                httpClient.start();
+                response = httpClient.newRequest("http://" + host + "/goform/Deviceinfo.xml")
+                        .timeout(3, TimeUnit.SECONDS).send();
+                int status = response.getStatus();
+                if (status == 200) {
+                    telnetEnable = false;
+                    logger.debug("We can access the HTTP API, disabling the Telnet mode by default");
+                }
+            } catch (Exception e) {
```

Comment:
See above: Please do not catch `Exception`

Reply:
See above, please suggest a solution for this.